lib/dockerRunner.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESSED OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+ * IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+'use strict'
+
+var Docker = require('dockerode')
+var parseUrl = require('parse-url')
+
+
+
+/**
+ * docker container runner for fuge
+ */
+module.exports = function () {
+  var docker = null
+
+  function connectToEngine (url) {
+    var connect = parseUrl(url)
+    var docker = null
+
+    if (connect.protocol === 'file') {
+      docker = new Docker({socketPath: connect.pathname})
+    } else if (connect.protocol === 'http') {
+      docker = new Docker({protocol: connect.protocol, host: connect.resource, port: connect.port})
+    }
+    return docker
+  }
```

Comment:
Passing in a URL is not necessary. `new Docker()` will connect using the system environment variables, which are more reliable than our config file.